supertokens_python/recipe/multifactorauth/multi_factor_auth_claim.py

```diff
@@ -0,0 +1,245 @@
+    MFARequirementList,
+    SessionRecipeUserIdInput,
+)
+from .utils import update_and_get_mfa_related_info_in_session
+
+
+class HasCompletedRequirementListSCV(SessionClaimValidator):
+    def __init__(
+        self,
+        id_: str,
+        claim: MultiFactorAuthClaimClass,
+        mfa_claim_validators: MultiFactorAuthClaimValidators,
+        requirement_list: MFARequirementList,
+        refetch_time_on_false_in_seconds: int,
+        max_age_in_seconds: Optional[int],
+    ):
+        super().__init__(id_)
+        # TDOD: fix type in SessionClaimValidator class with generic
+        self.claim: MultiFactorAuthClaimClass = claim  # type: ignore
+        self.mfa_claim_validators = mfa_claim_validators
+        self.refetch_time_on_false_in_ms = refetch_time_on_false_in_seconds * 1000
+        self.max_age_in_sec = max_age_in_seconds
+        self.requirement_list = requirement_list
+
+    async def validate(
+        self, payload: JSONObject, user_context: Dict[str, Any]
+    ) -> ClaimValidationResult:
+
+        if (self.claim.key not in payload) or (not payload[self.claim.key]):
+            return ClaimValidationResult(
+                is_valid=False, reason={"message": "Claim value not present in payload"}
+            )
+
+        claim_val: MFAClaimValue = payload[self.claim.key]
+
+        completed_factors = claim_val.c
+        next_set_of_unsatisfied_factors = (
+            self.claim.get_next_set_of_unsatisfied_factors(
+                completed_factors, self.requirement_list
+            )
+        )
+
+        if not next_set_of_unsatisfied_factors.factor_ids:
+            return ClaimValidationResult(is_valid=True)
+
+        factor_ids = next_set_of_unsatisfied_factors.factor_ids
+
+        if next_set_of_unsatisfied_factors.type == "string":
+            return ClaimValidationResult(
+                is_valid=False,
+                reason={
+                    "message": f"Factor validation failed: {factor_ids[0]} not completed",
+                    "factor_id": factor_ids[0],
+                },
+            )
+
+        elif next_set_of_unsatisfied_factors.type == "oneOf":
+            return ClaimValidationResult(
+                is_valid=False,
+                reason={
+                    "reason": f"None of these factors are complete in the session: {', '.join(factor_ids)}",
+                    "one_of": factor_ids,
+                },
+            )
+        else:
+            return ClaimValidationResult(
+                is_valid=False,
+                reason={
+                    "reason": f"Some of the factors are not complete in the session: {', '.join(factor_ids)}",
+                    "all_of_in_any_order": factor_ids,
+                },
+            )
+
+
+class HasCompletedRequirementListForAuthSCV(SessionClaimValidator):
+    def __init__(
+        self,
+        id_: str,
+        claim: MultiFactorAuthClaimClass,
+        mfa_claim_validators: MultiFactorAuthClaimValidators,
```

Comment:
if this is not used, you can remove mfa_claim_validators

Reply:
Done